### DISCUSSION TITLE
Problem to install gatsby CLI

### DISCUSSION THREAD

**Discussion by knupel** [gatsbyjs/gatsby#37631] [Help]
### Description

When I try to install the cli with the command `npm install -g gatsby-cli` I have this return.
It's write, I must be an administrator to do... and I'm the administrator of the account. So I don't understand why I don't have the permission to install by this way.

```
stanislasmarcais@macbook-pro-de-stanislas ~ % npm install -g gatsby-cli
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/lib/node_modules/gatsby-cli
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/gatsby-cli'
npm ERR!  [Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/gatsby-cli'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/usr/local/lib/node_modules/gatsby-cli'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stanislasmarcais/.npm/_logs/2023-02-08T15_18_18_766Z-debug-0.log
```

I don't know if it's a bug, or if I must remove security, because it's a global installion ?

I can install like that, but is is good ?
`brew install gatsby-cli` 

or there is another option to by pass the error ?

### Reproduction Link

/

### Steps to Reproduce

/

### Expected Result

/

### Actual Result

/

### Environment

```shell
Mac Book pro 2018 intel / OS Ventura 13.2 / Node 18.14
```

**Comment by gh640** [comment]
It's caused by the directory permission as the error message says. It's not specific to `gatsby-cli`, I guess.

You need to set the permissions properly before running the command. The following pages may help:

- https://stackoverflow.com/questions/49679808/error-eacces-permission-denied-mkdir-usr-local-lib-node-modules-node-sass-b
- https://github.com/npm/npm/issues/17268

**Comment by knupel** [comment]
Ok, I try to work on this way, thanks for the tips. But is possible just to don't intall de cli on local, not on global ?

**Comment by gh640** [comment]
I think there's a way for that. However, the issue is common and I recommend to resolve it. Solving it will save your time.

The tutorial says:

> If you are unable to successfully run the Gatsby CLI due to a permissions issue, you may want to check out the [npm docs on fixing permissions](https://docs.npmjs.com/getting-started/fixing-npm-permissions), or [this guide](https://github.com/sindresorhus/guides/blob/master/npm-global-without-sudo.md).

https://www.gatsbyjs.com/docs/tutorial/part-0/
